ksonnet-gen/ksonnet/emit.go

```diff
@@ -422,6 +429,20 @@ func (ao *apiObject) emitAsRefMixins(
 	m.writeLine("},")
 }
 
+func (ao *apiObject) emitConstructor(m *indentWriter) {
+	if dm, ok := ao.propertyMethods["default"]; ok {
+		log.Fatalf(
+			"Attempted to create constructor, but 'default' property already existed at '%s'",
+			dm.path)
+	}
+
+	if _, ok := ao.propertyMethods["metadata"]; !ok {
+		m.writeLine("default():: apiVersion + kind + {},")
+	} else {
+		m.writeLine("default(name):: apiVersion + kind + self.mixin.metadata.name(name),")
```

Comment:
There are a couple of problems with this:
1) Users may want to use `generateName`.  in that case, `name` must be blank.
2) It seems strange to have the name specified on things that often don't have names like "List" objects.  If we black out the metadata mixins we should probably black out name.